OlinkAnalyze/R/read_flex.R

```diff
@@ -0,0 +1,177 @@
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "ULOQ") |> 
+      dplyr::select(-SampleID, -tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC_Warning"))
+    
+    meta_data_per_assay <- dplyr::left_join(missfreq,norm_method, by = "OlinkID") |> 
+      dplyr::left_join(assay_warning, by = "OlinkID", multiple ="all") |>
+      dplyr::left_join(LLOQ, by = c("OlinkID")) |> 
+      dplyr::left_join(Plate_LQL, by = c("OlinkID", "PlateID1")) |> 
+      dplyr::left_join(LOD, by = c("OlinkID", "PlateID1")) 
+  } else {
+    LOD <- dat |> dplyr::filter(stringr::str_detect(SampleID, "LOD")) |> 
+      dplyr::select(-SampleID, -tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC_Warning")) |> 
+      tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                          names_to = "OlinkID",
+                          values_to = "LOD") 
+    meta_data_per_assay <- dplyr::left_join(missfreq,norm_method, by = c("OlinkID")) |> 
+      dplyr::left_join(LOD, by = c("OlinkID")) |> 
+      dplyr::select(-tidyselect::starts_with("PlateID"), -tidyselect::starts_with("QC_Warning"))
+  }
+  
+  
+  # pivot longer
+  dat<- dat |> 
+    dplyr::filter(!str_detect(SampleID, paste( "Missing Data freq.",
+                                               "Normalization",
+                                               "Assay warning",
+                                               "Lowest quantifiable level",
+                                               "Plate LOD",
+                                               "LLOQ",
+                                               "ULOQ",
+                                               "LOD", 
+                                               collapse = "|"))) |>  # remove end meta data
+    tidyr::pivot_longer(cols = tidyselect::starts_with("OID"),
+                        names_to = "OlinkID",
+                        values_to = ifelse(is_npx_data, "NPX", "Quantified_value")) |> 
+    dplyr::left_join(metadata, by = c("OlinkID")) |> 
+    dplyr::left_join(meta_data_per_assay, multiple = "all") |> 
+    dplyr::mutate(PlateID = PlateID1) |> 
+    dplyr::mutate(QC_Warning = QC_Warning1) |> 
+    dplyr::select(SampleID, OlinkID,
+                  UniProt, Assay, MissingFreq,
+                  Panel,PlateID,
+                  QC_Warning,dplyr::matches("Plate_LQL"),
+                  dplyr::matches("LOD"),
+                  dplyr::matches("Plat_LOD"),
+                  dplyr::matches("LLOQ"),
+                  dplyr::matches("ULOQ"),
+                  dplyr::matches("NPX"),
+                  dplyr::matches("Quantified_value"),
+                  dplyr::matches("Unit"),
+                  dplyr::matches("Assay_Warning"),
+                  dplyr::matches("Normalization"),
+                  dplyr::matches("*Inc Ctrl*"),
+                  dplyr::matches("*Det Ctrl*"))
```

Comment:
Maybe could be simplified with `tidyselect::any_of()`?

Also, if im not mistaken *Inc Ctrl* and *Det Ctrl* are never set with current implemenetation...